frontend/model/state.js

```diff
@@ -8,42 +8,53 @@ import Vue from 'vue'
 import Vuex from 'vuex'
 import { GIMessage } from '~/shared/GIMessage.js'
 import * as _ from '@utils/giLodash.js'
+import L from '@view-utils/translations.js'
 import { SETTING_CURRENT_USER } from './database.js'
+import { ErrorDBMalformed, ErrorDBConnection } from '~/shared/domains/gi/db.js'
 import { LOGIN, LOGOUT, EVENT_HANDLED, CONTRACTS_MODIFIED } from '@utils/events.js'
 import Colors from './colors.js'
+import { TypeValidatorError } from '@utils/flowTyper.js'
+import { GIErrorUnrecoverable, GIErrorIgnoreAndBanIfGroup, GIErrorDropAndReprocess } from './errors.js'
+import { STATUS_OPEN, PROPOSAL_REMOVE_MEMBER } from './contracts/voting/proposals.js'
+import { VOTE_FOR } from '@model/contracts/voting/rules.js'
+import { WE_JUST_JOINED } from '@model/constants.js'
 import './contracts/group.js'
 import './contracts/mailbox.js'
 import './contracts/identity.js'
 
-// babel transforms lodash imports: https://github.com/lodash/babel-plugin-lodash#faq
-// for diff between 'lodash/map' and 'lodash/fp/map'
-// see: https://github.com/lodash/lodash/wiki/FP-Guide
-
 Vue.use(Vuex)
 var store // this is set and made the default export at the bottom of the file.
 // we have it declared here to make it accessible in mutations
-// 'state' is the Vuex state object
-// NOTE: THE STATE CAN ONLY STORE *SERIALIZABLE* OBJECTS! THAT MEANS IF YOU TRY
-//       TO STORE AN INSTANCE OF A CLASS (LIKE A CONTRACT), IT WILL NOT STORE
-//       THE ACTUAL CONTRACT, BUT JSON.STRINGIFY(CONTRACT) INSTEAD!
+// 'state' is the Vuex state object, and it can only store JSON-like data
+var dropAllMessagesUntilRefresh = false
 
 const CONTRACT_REGEX = /^gi\.contracts\/(?:([^/]+)\/)(?:([^/]+)\/)?process$/
 // guard all sbp calls for contract actions with this function
 export function selectorIsContractOrAction (sel: string) {
   if (!CONTRACT_REGEX.test(sel)) {
-    throw new Error(`bad selector '${sel}' for contract type!`)
+    throw new GIErrorIgnoreAndBanIfGroup(`bad selector '${sel}' for contract type!`)
   }
 }
 
 sbp('sbp/selectors/register', {
-  // This will build the current contract state from applying all its actions
   'state/latestContractState': async (contractID: string) => {
     let events = await sbp('backend/eventsSince', contractID, contractID)
     events = events.map(e => GIMessage.deserialize(e))
-    const state = {}
+    var state = {}
```

Comment:
when we need to re-assign a variable use [`let`](https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Statements/let).